File: Desafio_Rivia/2_csv_to_json.js
const fs = require('fs');
const csv = require('csv-parser');

// Função para converter CSV em JSON
function csvParaJson(csvArquivo, jsonArquivo) {
    const resultados = [];

    // Lê o arquivo CSV e converte para um array de objetos
    fs.createReadStream(csvArquivo)
        .pipe(csv())
        .on('data', (dados) => resultados.push(dados))  // Armazena cada linha do CSV como objeto
        .on('end', () => {
            // Escreve o array de objetos no arquivo JSON
            fs.writeFileSync(jsonArquivo, JSON.stringify(resultados, null, 4), 'utf8');
            console.log(`Arquivo JSON gerado: ${jsonArquivo}`);
        });
}

// Exemplo de uso
csvParaJson('dados.csv', 'dados.json');

File: Desafio_Rivia/2_csv_to_json.py
import csv
import json

def csv_para_json(csv_arquivo, json_arquivo):
    dados = []
    with open(csv_arquivo, mode='r') as arquivo_csv:
        csv_reader = csv.DictReader(arquivo_csv)
        for linha in csv_reader:
            dados.append(linha)

    with open(json_arquivo, mode='w') as arquivo_json:
        json.dump(dados, arquivo_json, indent=4)

# Exemplo de uso
csv_para_json('dados.csv', 'dados.json')
